crates/sui-core/src/authority.rs

```diff
@@ -1570,9 +1578,21 @@ impl ExecutionState for AuthorityState {
                     // to persist the consensus index. If the validator crashes, this transaction
                     // may be resent to the checkpoint logic that will simply ignore it.
 
-                    // Cache the next checkpoint number if it changes
+                    // Cache the next checkpoint number if it changes.
                     self.latest_checkpoint_num
                         .store(checkpoint.next_checkpoint(), Ordering::Relaxed);
+
+                    // TODO: At this point we should know whether we want to change epoch. If we do,
+                    // we should have (i) the new committee and (ii) the new keypair of this authority.
+                    // We then call:
+                    // ```
+                    //  self
+                    //      .tx_reconfigure_consensus
+                    //      .send((new_keypair, new_committee))
+                    //      .await
+                    //      .expect("Failed to reconfigure consensus");
+                    // ```
+                    let _tx_reconfigure_consensus = &self.tx_reconfigure_consensus;
```

Comment:
I'm not sure where it is but I gotta think there is a different place where we are doing epoch changes and notify.  This comment is partly so that others more knowledgeable about epoch changes, such as @lxfind can enlighten where it is in the code path.  I'd like to put in an event for epoch changes.

Reply:
This is a mpsc sender so it is clonal. The AuthorityState holds the sender so that we can use the channel whenever it is needed.